### PR TITLE
GotBot Optimized(Hopefully)

### PR DESCRIPTION
**Pull request by kaushiksk**
What has been done.

- Data preprocessing is only done once now by running gotpreprocess.py.
- While training saved preprocessed data is loaded instead of preprocessing each time.
- Added a dropout Layer to the Neural Net. Will Hopefully improve performance.
- Vocabulary consists of the entire test. So during training no <unknown>s will be encountered.

To do

- [ ] Test if the model improves in performance
- [ ] Test the model in --test-interactive mode (Haven't done this, might throw unchecked errors)
- [ ] Run the model on test data set when user interrupts training.
- [ ] Test performance by removing uncommon words. This just requires a small modification in gotpreprocess.py.
- [ ] Try attention_seq2seq
- [ ] Try with different embedding size. ( I changed the current one from 32 to 300, this will slow down training but I'm hoping it improves final performance )

### CODE REVIEW

**Comment by kaushiksk** [comment]
@vshan and everyone, please train the model in your systems and see if it actually performs better. Mine takes forever on a 4GB ram :p

**Comment by vshan** [comment]
@kaushiksk awesome work kaushik. Will go through this.
